src/Microsoft.Azure.NotificationHubs/Messaging/EntityDescriptionSerializer.cs

```diff
@@ -160,7 +163,7 @@ public string Serialize(EntityDescription description)
                 description = new GcmTemplateRegistrationDescription((FcmTemplateRegistrationDescription) description);
             }
 
-            var serializer = GetSerializer(description.GetType().Name);
+            var serializer = GetSerializer(typeof(RegistrationDescription).Name);
```

Comment:
It seems like we would want to make this change to both `Serialize()` methods.

But, how does this change affect the wire representation of the type? And is that compatible with the REST API?

Reply:
Yep, It was midnight proof of concept. Second serializer also fixed.

We will serialize all kind of descriptions to `<RegistrationDescription>` with `i:type` attribute(exactly what we need). And deserialize in proper type independently of incoming message format.
It's fully compatible with API. We tested this changes with `Samples` and example from docs as well.

Reply:
1. Can this represent the same XML? The XML needs to of structure : <RegistrationDescription type="WindowsRegistrationDescription" .... />
2. if it working yo need to update deserialze as well. 
3. What is the significance of having other serializers then ?

Reply:
Do you mean `WindowsRegistrationDescription.Serialize()`? We can create this description with NotificationHubClient.CreateWindowsNativeRegistrationAsync method, so we use this serializer. The same thing with other serializers registered in `EntityDescriptionSerializer `.